crates/rspack_plugin_copy/src/lib.rs

```diff
@@ -597,8 +622,64 @@ fn escape_glob_chars(s: &str) -> String {
   escaped
 }
 
+fn set_info(target: &mut AssetInfo, info: Info) {
+  if let Some(minimized) = info.minimized {
+    target.minimized = minimized;
+  }
+
+  if let Some(immutable) = info.immutable {
+    target.immutable = immutable;
+  }
+
+  if let Some(chunk_hash) = info.chunk_hash {
+    target.chunk_hash = rustc_hash::FxHashSet::from_iter(chunk_hash.into_iter());
+  }
+
+  if let Some(content_hash) = info.content_hash {
+    target.content_hash = rustc_hash::FxHashSet::from_iter(content_hash.into_iter());
+  }
+
+  if let Some(development) = info.development {
+    target.development = development;
+  }
+
+  if let Some(hot_module_replacement) = info.hot_module_replacement {
+    target.hot_module_replacement = hot_module_replacement;
+  }
+
+  if let Some(related) = info.related {
+    target.related = AssetInfoRelated {
+      source_map: related.source_map,
+    };
+  }
+
+  if let Some(version) = info.version {
+    target.version = version;
+  }
+}
+
 #[test]
 fn test_escape() {
   assert_eq!(escape_glob_chars("a/b/**/*.js"), r#"a/b/\*\*/\*.js"#);
   assert_eq!(escape_glob_chars("a/b/c"), r#"a/b/c"#);
 }
+
+// If this test fails, you should modify `set_info` function, according to your changes about AssetInfo
+// Make sure every field of AssetInfo is considered
+#[test]
+fn ensure_info_fields() {
```

Comment:
Why can't just use AssetInfo

Reply:
User specify the info config, so user can specify any field they want, for the other fields, stay untouched, so users info fields are all optional